### PR TITLE
Brad/add strict and complexity

### PR DESCRIPTION
**Pull request by bbraithwaite**
Adds: 

* Complexity rules
* The newer form of padding 
* Enforces 'use strict' in function closures

### CODE REVIEW

File: rules/stylistic-issues.js [hunk 1]
```diff
@@ -188,8 +188,19 @@ module.exports = {
         // require quotes around object literal property names
         "quote-props": ["error", "as-needed", { keywords: false, unnecessary: true, numbers: false }],
 
-        // require or disallow an empty line after variable declarations
-        "newline-after-var": ["error", "always"]
+        // enforce a maximum depth that blocks can be nested
+        "max-depth": ["error", 4],
+
+        // enforce a maximum number of parameters in function definitions
+        "max-params": ["error", 3],
```
Comment (bbraithwaite) [inline rules/stylistic-issues.js:195]: //cc @laurengarcia @MariagraziaAlastra

File: rules/stylistic-issues.js [hunk 2]
```diff
@@ -188,8 +188,19 @@ module.exports = {
         // require quotes around object literal property names
         "quote-props": ["error", "as-needed", { keywords: false, unnecessary: true, numbers: false }],
 
-        // require or disallow an empty line after variable declarations
-        "newline-after-var": ["error", "always"]
+        // enforce a maximum depth that blocks can be nested
+        "max-depth": ["error", 4],
```
Comment (bbraithwaite) [inline rules/stylistic-issues.js:192]: //cc @laurengarcia @MariagraziaAlastra

Comment (andrey-p) [reply rules/stylistic-issues.js:192]: :+1:

Might need to increase it a bit, pretty much all our code is in at least two blocks by default (e.g. IIFE, constructor).

File: rules/stylistic-issues.js (hunk 1 above)
Comment (laurengarcia) [reply rules/stylistic-issues.js:195]: lgtm

File: rules/stylistic-issues.js (hunk 2 above)
Comment (laurengarcia) [reply rules/stylistic-issues.js:192]: lgtm

**Pull request merged**